### ISSUE TITLE
First long header is not recognized as an error

### ISSUE THREAD

**Issue by emilyseville7cfg** [DavidAnson/markdownlint#433]
Hello! I have [README](https://github.com/Console-Utils/awk-gendocs-util/tree/ci#readme) file and the following .yml config:
```yml
# MD010/no-hard-tabs - Hard tabs
MD010:
  # Number of spaces for each hard tab
  spaces_per_tab: 2

# MD013/line-length - Line length
MD013:
  # Number of characters
  line_length: 200
  # Number of characters for headings
  heading_line_length: 10
  # Number of characters for code blocks
  code_block_line_length: 80

# MD024/no-duplicate-heading/no-duplicate-header - Multiple headings with the same content
MD024:
  # Only check sibling headings
  siblings_only: true

# MD026/no-trailing-punctuation - Trailing punctuation in heading
MD026:
  # Punctuation characters
  punctuation: ".,;:!?"

# MD036/no-emphasis-as-heading/no-emphasis-as-header - Emphasis used instead of a heading
MD036:
  # Punctuation characters
  punctuation: ".,;:!?"

# MD043/required-headings/required-headers - Required heading structure
MD043:
  # List of headings
  headings:
    - "+"
    - "## Description"
    - "+"
    - "## Syntax"
    - "## Return codes"
    - "*"
```
I don't understand why there is no linter error for `## Requirenments` heading about long length exceeding 10 chars.

**Comment by DavidAnson** [comment]
The default behavior of MD013 will not report a violation if there are no spaces beyond the configured limit. In other words, if there's not an easy change to make the line shorter, it allows the line through. If you want to be strict about the length limit, the "strict" option should do what you want. You may also be interested in the "stern" option, though it wouldn't help in this particular case. The rule documentation has more about this and examples: https://github.com/DavidAnson/markdownlint/blob/main/doc/Rules.md#md013
